### PR TITLE
Optimize configs

### PR DESCRIPTION
**Pull request by NanoMichael**
Optimized configurations, make auto font find optional (since WASM doesn't need this feature, that can reduce wasm file size).

### CODE REVIEW

File: lib/meson.build [hunk 1]
```diff
@@ -39,6 +39,18 @@ src += utils_src
 subdir('render')
 src += render_src
 
+configure_file(
+    input: 'config.h.in',
+    output: 'config.h',
+    configuration: {
+        'PROJECT_VERSION_MAJOR': 1,
+        'PROJECT_VERSION_MINOR': 0,
+        'PROJECT_VERSION_PATCH': 0,
```
Comment (NanoMichael) [inline lib/meson.build:48]: It seems meson does not have functions like `meson.project_version_{major/minor/patch}`, maybe @sp1ritCS could help.

**Comment by sp1ritCS** [comment]
have you seen the comment I made about the soversion of meson library() invocation?

**Comment by NanoMichael** [comment]
> have you seen the comment I made about the soversion of meson library() invocation?

No, and I have not received any emails.

**Comment by NanoMichael** [comment]
> have you seen the comment I made about the soversion of meson library() invocation?

Oh, I've forgotten this, will fix it later.

**Comment by NanoMichael** [comment]
> Could you rename the config.h to a more unique name?

Sure, I'll rename it to `tinytexconfig.h`

**Pull request merged**